### PR TITLE
feat(definition-list,select): describe DefinitionList & rest of selec…

### PR DESCRIPTION
**Pull request by mjaniczko-sage**
…t component using design tokens

### Proposed behaviour
Describes the Definition List component using design tokens.
Remove last theme properties from Select Component
Update test after above changes

### Current behaviour
Component use the theme styled-component property.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Additional context
All necessary informations: FE-4931

### Testing instructions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

### CODE REVIEW

**Review by mkrds — APPROVED** [review]

**Comment by mkrds** [comment]
All good but you made typos in the commit message

File: src/components/definition-list/definition-list.style.js [hunk 1]
```diff
@@ -40,39 +39,27 @@ export const StyledDdDiv = styled.div`
...
+    a, button {
+      font-weight: 700px;
+      text-decoration: none;
     }
-  `}
+  }
   ${space}
 `;
-
-StyledDd.defaultProps = {
```
Comment (JoannaMikul) [inline src/components/definition-list/definition-list.style.js:76]: StyledDd use space so you shouldn't remove this.

File: src/components/definition-list/definition-list.style.js [hunk 2]
```diff
@@ -40,39 +39,27 @@ export const StyledDdDiv = styled.div`
...
-  theme: baseTheme,
-};
-
 export const StyledDd = styled.dd`
-  ${({ theme }) => css`
-    font-size: 14px
-    font-weight: 700;
-    color: ${theme.definitionList.ddText};
-    margin-left: 0px;
+  font-size: 14px
```
Comment (JoannaMikul) [inline src/components/definition-list/definition-list.style.js:48]: no semi-colon at the end

**Review by jhajduk-sage — DISMISSED** [review]

**Review by mkrds — DISMISSED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by divyajindel — DISMISSED** [review]

**Review by divyajindel — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by mkrds — APPROVED** [review]

**Pull request merged**